packages/@sanity/codegen/src/typescript/findQueriesInSource.ts

```diff
@@ -41,12 +42,18 @@ export function findQueriesInSource(
       const init = node.init
 
       // Look for tagged template expressions that are called with the `groq` tag
-      if (
+      const isGroqTemplateTag =
         babelTypes.isTaggedTemplateExpression(init) &&
         babelTypes.isIdentifier(init.tag) &&
-        babelTypes.isIdentifier(node.id) &&
         init.tag.name === groqTagName
-      ) {
+
+      // Look for strings wrapped in a defineQuery function call
+      const isDefineQueryCall =
+        babelTypes.isCallExpression(init) &&
+        babelTypes.isIdentifier(init.callee) &&
+        init.callee.name === defineQueryFunctionName
```

Comment:
since `defineQuery` is more ambiguous, and also exported by other packages, we should validate that it's the  `defineQuery` from `groq`

Reply:
I have added a helper to check where the `defineQuery` identifier actually comes from.

I implemented it only for the following import styles.

```ts
import { defineQuery } from "groq"
const query = defineQuery(" ... ")
```

```ts
import * as groq from "groq"

// Note: this is not supported by resolveExpression
const query = groq.defineQuery("...")
```

```ts
const { defineQuery } = require("groq")

// Note: resolveExpression works here, but is not set up to support commonjs 
// when detecting query fragments
const query = defineQuery("...")
```

Notably, I did not implement it for 
```ts
const groq = require('groq')
const query = groq.defineQuery("...")
```
as that would add significant complexity and is not supported by `resolveExpression`.

Let me know if you would like me to add that case too.

Reply:
Perfect! We don't need to support all variants, we can add more as we see them.
Member expressions aren't supported, no - but that should be handled separately anyway